### PR TITLE
Chore: refactor

### PR DESCRIPTION
**Pull request by foxpy**

### CODE REVIEW

File: src/helpers/cosmos.ts [hunk 1]
```diff
@@ -110,15 +109,16 @@ export class CosmosWrapper {
     const txhash = res.data?.tx_response.txhash;
     while (numAttempts > 0) {
       await waitBlocks(this.sdk, 1);
-      try {
-        numAttempts--;
-        const data = await rest.tx.getTx(this.sdk as CosmosSDK, txhash);
+      numAttempts--;
+      const data = await rest.tx
+        .getTx(this.sdk as CosmosSDK, txhash)
+        .catch(() => null);
```
Comment (ratik) [inline src/helpers/cosmos.ts:115]: mb we can store error and re-throw it at 121?

File: src/testcases/interchain_kv_query.test.ts [hunk 2]
```diff
@@ -465,7 +458,7 @@ describe('Neutron / Interchain KV Query', () => {
       );
       for (
         let i = 0;
-        i < max([2, 3, 4].map((i) => query[i].updatePeriod)) + 1;
+        i < max([2, 3, 4].map((i) => updatePeriods[i])) + 1;
```
Comment (ratik) [inline src/testcases/interchain_kv_query.test.ts:461]:
```suggestion
        i <= max(Object.values(updatePeriods));
```

File: src/testcases/interchain_kv_query.test.ts [hunk 3]
```diff
@@ -171,18 +171,17 @@ const validateBalanceQuery = async (
...
-// TODO: throw away this type alltogether
-type Query = {
-  updatePeriod: number;
-};
-
 describe('Neutron / Interchain KV Query', () => {
+  const connectionId = 'connection-0';
+  const updatePeriods: { [key: number]: number } = {
+    2: 2,
+    3: 4,
```
Comment (NeverHappened) [inline src/testcases/interchain_kv_query.test.ts:178]: typo?: `3: 3, 4: 4`

Comment (foxpy) [reply src/testcases/interchain_kv_query.test.ts:178]: It doesn't make any sense actually, it is just a pair of query id and update period

File: src/helpers/ica.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ * getIca queries interchain account address
+ */
+export const getIca = async (
+  cm: CosmosWrapper,
+  contractAddress: string,
+  icaId: string,
+  connectionId: string,
+) => {
+  await waitForIca(cm, contractAddress, icaId, connectionId);
```
Comment (NeverHappened) [inline src/helpers/ica.ts:39]: It seems we always doing double query here: wait for result, and then get same result.
I propose to refactor `waitWithAttempts` func to something like this:
```
// getWhenReady tries to get results until the data is ready
export const getWhenReady = async <T>(
  sdk: any,
  getFunc: () => Promise<T>,
  readyFunc: (t: T) => boolean,
  numAttempts = 20,
): Promise<T> => {
  while (numAttempts > 0) {
    numAttempts--;

    try {
      const data = await getFunc();
      if (readyFunc(data)) {
        return data;
      }
      await waitBlocks(sdk, 1);
    } catch {
      continue;
    }
  }
  throw new Error('getWhenReady: no attempts left');
};
```

I didn't test it!
But this allows to separate check if ready from getting and to return result simultaneously.

Comment (foxpy) [reply src/helpers/ica.ts:39]: That means I will have to redo `waitWithAttempts()` somehow to make its predicate not only return completion status, but also return queried data.

I don't have any ideas on how to do that without writing too much code, and tests wouldn't really speed up after saving 1 ms or less, so it doesn't seem to be worthy for me to refactor these functions.

They are easy to read and they work well enough.

Unless you have an idea on how to do that cleanly?

Comment (foxpy) [reply src/helpers/ica.ts:39]: oops, sorry, I didn't read your snippet thoroughly :facepalm: 

I will investigate if it is possible to refactor the waiting functions

File: src/testcases/interchaintx.test.ts [hunk 5]
```diff
@@ -406,20 +281,20 @@ describe('Neutron / Interchain TXs', () => {
...
+      await getWithAttempts(
         cm1.sdk,
-        async () =>
-          (
-            await cm1.listIBCChannels()
-          ).channels.filter((c) => c.channel_id == 'channel-3')[0].state ==
-          'STATE_OPEN',
+        () => cm1.listIBCChannels(),
+        (channels) =>
+          channels.channels.filter((c) => c.channel_id == 'channel-3')[0]
```
Comment (NeverHappened) [inline src/testcases/interchaintx.test.ts:296]: There is `array.find` function that you can use

File: src/testcases/interchain_tx_query.test.ts [hunk 6]
```diff
@@ -297,11 +302,17 @@ describe('Neutron / Interchain TX Query', () => {
       const res = await cm2.msgSend(watchedAddr3, amountToAddr3_2.toString());
       expectedIncomingTransfers++;
       expect(res.code).toEqual(0);
+      // initiate query before relayer has any chance to submit query data
+      const depositsPromise = queryRecipientTxs(
```
Comment (NeverHappened) [inline src/testcases/interchain_tx_query.test.ts:306]: I don't understand why can't we move this to line 315 where we wait for it?

Comment (foxpy) [reply src/testcases/interchain_tx_query.test.ts:314]: it literally says `// initiate query before relayer has any chance to submit query data`

File: README.md [hunk 7]
```diff
@@ -36,7 +38,8 @@ yarn test:interchain_tx_query # interchain tx query test
 yarn test:interchain_kv_query # interchain tx query test
 ```
 
-NOTE: To speed up tests you can run the cosmopark by youself with `make start-cosmopark` in the setup dir. To run test with the already running cosmopark use `NO_DOCKER` env variable.
+NOTE: To speed up tests you can run the cosmopark by youself with `make start-cosmopark` in the setup dir. To run test
+with the already running cosmopark use `NO_DOCKER` env variable.
```
[the remark this answers is not in the document]
Comment (foxpy) [reply README.md:42]: oops

[the remark this answers is not in the document]
Comment (foxpy) [reply README.md:42]: actually not oops, seems like it has been automatically capped at 120 characters per line.

[the remark this answers is not in the document]
Comment (sotnikov-s) [reply README.md:42]: Wait, we have even longer lines in other repos. It means the linter (if it's the linter who's done it) doesn't know its place and tries to control `.md` files whereas it's meant to take care about `.ts` files only. The interesting fact is that the lines are the same they are in main, but only in your PR they've been changed, means it's most likely your linter settings that's made it. If it's not the linter, then IDE maybe, but anyway it seems redundant

[the remark this answers is not in the document]
Comment (foxpy) [reply README.md]: That was my IDE. I have canceled these changes.

**Review by oopcode — APPROVED** [review]

**Review by sotnikov-s — APPROVED** [review]

**Pull request merged**